File: src/main/java/org/manalith/irc/ManalithIRC.java
package org.manalith.irc;

import org.apache.commons.configuration.ConfigurationException;
import org.apache.commons.configuration.XMLConfiguration;
import org.apache.log4j.Logger;
import org.manalith.irc.ui.ApplicationWindow;

/**
 * ManalithIRC Launcher
 * 
 * @author setzer
 * 
 */
public class ManalithIRC {
	private static final Logger logger = Logger.getLogger(ManalithIRC.class);
	public static Application application;

	/**
	 * @param args
	 */
	public static void main(String[] args) {
		try {
			XMLConfiguration config = new XMLConfiguration("config.xml");
			application = new Application(config);
			ApplicationWindow.load();
		} catch (ConfigurationException e) {
			logger.error(e);
		}
	}
}

File: src/main/scala/org/manalith/irc/ManalithIRC.scala
package org.manalith.irc;

import org.apache.commons.configuration.ConfigurationException
import org.apache.commons.configuration.XMLConfiguration
import org.manalith.irc.helper.LogHelper
import org.manalith.irc.model.ConnectionManager
import org.manalith.irc.model.Preferences
import org.manalith.irc.ui.ApplicationWindowLoader

/**
 * ManalithIRC Launcher
 *
 * @author setzer
 *
 */
object ManalithIRC extends LogHelper {
	/**
	 * @param args
	 */
	def main(args: Array[String]) {
		try {
			Preferences.config = new XMLConfiguration("config.xml");
			ApplicationWindowLoader.load;
		} catch {
			case e: ConfigurationException =>
				logger.error(e);
		}
	}

	def onExit() {
		ConnectionManager.disconnectAllConnection(Preferences.messages_disconnect);
	}
}
